test/e2e_node/device_plugin.go

```diff
@@ -0,0 +1,268 @@
+				framework.ExpectNoError(err)
+				return numberOfDevices(node, resourceName) <= 0
+			}, 10*time.Minute, framework.Poll).Should(BeTrue())
+
+			By("Checking that scheduled pods can continue to run even after we delete device plugin.")
+			count1, devIdRestart1 = parseLogFromNRuns(f, pod1.Name, pod1.Name, count1+1, deviceIDRE)
+			Expect(devIdRestart1).To(Equal(devId1))
+			count2, devIdRestart2 := parseLogFromNRuns(f, pod2.Name, pod2.Name, count2+1, deviceIDRE)
+			Expect(devIdRestart2).To(Equal(devId2))
+
+			By("Restarting Kubelet.")
+			restartKubelet()
+
+			By("Checking that scheduled pods can continue to run even after we delete device plugin and restart Kubelet.")
+			count1, devIdRestart1 = parseLogFromNRuns(f, pod1.Name, pod1.Name, count1+2, deviceIDRE)
+			Expect(devIdRestart1).To(Equal(devId1))
+			count2, devIdRestart2 = parseLogFromNRuns(f, pod2.Name, pod2.Name, count2+2, deviceIDRE)
+			Expect(devIdRestart2).To(Equal(devId2))
+
+			// Cleanup
+			f.PodClient().DeleteSync(pod1.Name, &metav1.DeleteOptions{}, framework.DefaultPodDeletionTimeout)
+			f.PodClient().DeleteSync(pod2.Name, &metav1.DeleteOptions{}, framework.DefaultPodDeletionTimeout)
+		})
+	})
+})
+
+// makeBusyboxPod returns a simple Pod spec with a busybox container
+// that requests resourceName and runs the specified command.
+func makeBusyboxPod(resourceName, cmd string) *v1.Pod {
+	podName := "device-plugin-test-" + string(uuid.NewUUID())
+	rl := v1.ResourceList{v1.ResourceName(resourceName): *resource.NewQuantity(1, resource.DecimalSI)}
+
+	return &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: podName},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyAlways,
+			Containers: []v1.Container{{
+				Image: busyboxImage,
+				Name:  podName,
+				// Runs the specified command in the test pod.
+				Command: []string{"sh", "-c", cmd},
+				Resources: v1.ResourceRequirements{
+					Limits:   rl,
+					Requests: rl,
+				},
+			}},
+		},
+	}
+}
+
+// parseLogFromNRuns returns restart count of the specified container
+// after it has been restarted at least restartCount times,
+// and the matching string for the specified regular expression parsed from the container logs.
+func parseLogFromNRuns(f *framework.Framework, podName string, contName string, restartCount int32, re string) (int32, string) {
+	var count int32
+	// Wait till pod has been restarted at least restartCount times.
+	Eventually(func() bool {
+		p, err := f.PodClient().Get(podName, metav1.GetOptions{})
+		if err != nil || len(p.Status.ContainerStatuses) < 1 {
+			return false
+		}
+		count = p.Status.ContainerStatuses[0].RestartCount
+		return count >= restartCount
+	}, 5*time.Minute, framework.Poll).Should(BeTrue())
+	logs, err := framework.GetPodLogs(f.ClientSet, f.Namespace.Name, podName, contName)
+	if err != nil {
+		framework.Failf("GetPodLogs for pod %q failed: %v", podName, err)
+	}
+	framework.Logf("got pod logs: %v", logs)
```

Comment:
NIT: maybe add some line returns in the function :)

Reply:
Sorry, what line do you mean?

Reply:
Any line :) Adding some line returns would seem more consistent with your coding style

Reply:
Thanks, done:)